### PR TITLE
Migrate FilePanel to TypeScript

### PR DESCRIPTION
**Pull request by gsouquet**
Migrated as part of my investigation for vector-im/element-web#17753

Requires matrix-org/matrix-react-sdk#6283

### CODE REVIEW

File: src/components/structures/FilePanel.tsx [hunk 1]
```diff
@@ -16,9 +16,12 @@ limitations under the License.
 */
 
 import React from 'react';
-import PropTypes from 'prop-types';
 
 import {Filter} from 'matrix-js-sdk/src/filter';
```
Comment (t3chguy) [inline src/components/structures/FilePanel.tsx:20]:
```suggestion
import { Filter } from 'matrix-js-sdk/src/filter';
```

File: src/components/structures/FilePanel.tsx [hunk 2]
```diff
@@ -16,9 +16,12 @@ limitations under the License.
...
+import { EventTimelineSet } from "matrix-js-sdk/src/models/event-timeline-set";
+import { MatrixEvent, Room } from 'matrix-js-sdk/src';
+import { TimelineWindow } from 'matrix-js-sdk/src/timeline-window';
+
 import * as sdk from '../../index';
 import {MatrixClientPeg} from '../../MatrixClientPeg';
```
Comment (t3chguy) [inline src/components/structures/FilePanel.tsx:26]:
```suggestion
import { MatrixClientPeg } from '../../MatrixClientPeg';
```

File: src/components/structures/FilePanel.tsx [hunk 3]
```diff
@@ -28,25 +31,33 @@ import {RightPanelPhases} from "../../stores/RightPanelStorePhases";
 import DesktopBuildsNotice, {WarningKind} from "../views/elements/DesktopBuildsNotice";
 import {replaceableComponent} from "../../utils/replaceableComponent";
```
Comment (t3chguy) [inline src/components/structures/FilePanel.tsx:32]:
```suggestion
import DesktopBuildsNotice, { WarningKind } from "../views/elements/DesktopBuildsNotice";
import { replaceableComponent } from "../../utils/replaceableComponent";
```

and RightPanelPhases above but it won't let me comment on it :(

File: src/components/structures/FilePanel.tsx [hunk 4]
```diff
@@ -28,25 +31,33 @@ import {RightPanelPhases} from "../../stores/RightPanelStorePhases";
...
```
Comment (t3chguy) [inline src/components/structures/FilePanel.tsx:33]: random line break?

File: src/components/structures/FilePanel.tsx [hunk 5]
```diff
@@ -28,25 +31,33 @@ import {RightPanelPhases} from "../../stores/RightPanelStorePhases";
...
-    decryptingEvents = new Set();
+    private decryptingEvents = new Set<string>();
+    public noRoom: boolean;
 
     state = {
         timelineSet: null,
     };
 
-    onRoomTimeline = (ev, room, toStartOfTimeline, removed, data) => {
+    private onRoomTimeline = (ev: MatrixEvent, room: Room, toStartOfTimeline: true, removed: true, data: any) => {
```
Comment (t3chguy) [inline src/components/structures/FilePanel.tsx:60]: `: void` ?

File: src/components/structures/FilePanel.tsx [hunk 6]
```diff
@@ -60,7 +71,7 @@ class FilePanel extends React.Component {
         }
     };
 
-    onEventDecrypted = (ev, err) => {
+    private onEventDecrypted = (ev: MatrixEvent, err?: any) => {
```
Comment (t3chguy) [inline src/components/structures/FilePanel.tsx:74]: `: void` ?

File: src/components/structures/FilePanel.tsx [hunk 7]
```diff
@@ -70,7 +81,7 @@ class FilePanel extends React.Component {
         this.addEncryptedLiveEvent(ev);
     };
 
-    addEncryptedLiveEvent(ev, toStartOfTimeline) {
+    public addEncryptedLiveEvent(ev: MatrixEvent) {
```
Comment (t3chguy) [inline src/components/structures/FilePanel.tsx:84]: `: void` ?

File: src/components/structures/FilePanel.tsx [hunk 8]
```diff
@@ -117,7 +128,7 @@ class FilePanel extends React.Component {
         }
     }
 
-    async fetchFileEventsServer(room) {
+    public async fetchFileEventsServer(room: Room) {
```
Comment (t3chguy) [inline src/components/structures/FilePanel.tsx:131]: `: Promise<void>` ?

File: src/components/structures/FilePanel.tsx [hunk 9]
```diff
@@ -16,9 +16,12 @@ limitations under the License.
...
 
 import React from 'react';
-import PropTypes from 'prop-types';
 
 import {Filter} from 'matrix-js-sdk/src/filter';
+import { EventTimelineSet } from "matrix-js-sdk/src/models/event-timeline-set";
+import { MatrixEvent, Room } from 'matrix-js-sdk/src';
+import { TimelineWindow } from 'matrix-js-sdk/src/timeline-window';
+
 import * as sdk from '../../index';
```
Comment (t3chguy) [inline src/components/structures/FilePanel.tsx:25]: would be good to kill this off during this conversion imo

File: src/components/structures/FilePanel.tsx [hunk 10]
```diff
@@ -16,9 +16,12 @@ limitations under the License.
 */
 
 import React from 'react';
-import PropTypes from 'prop-types';
 
 import {Filter} from 'matrix-js-sdk/src/filter';
+import { EventTimelineSet } from "matrix-js-sdk/src/models/event-timeline-set";
+import { MatrixEvent, Room } from 'matrix-js-sdk/src';
```
Comment (t3chguy) [inline src/components/structures/FilePanel.tsx:22]: seems strange to directly import some and import some from the index, one or the other please

**Review by t3chguy — APPROVED** [review]
LGTM otherwise

File: src/components/structures/FilePanel.tsx (hunk 9 above)
Comment (gsouquet) [reply src/components/structures/FilePanel.tsx:26]: As far as I'm aware it's still needed for tests sometimes?
Under what circumstances is it acceptable to remove?

Comment (t3chguy) [reply src/components/structures/FilePanel.tsx:26]: Well this'll be for the `getComponent` calls below, my point was we can probably eradicate them

**Pull request merged**